functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -211,49 +254,103 @@ func (p *HelmChartInflationGeneratorPlugin) cleanup() {
 	if p.tmpDir != "" {
 		os.RemoveAll(p.tmpDir)
 	}
+	if strings.HasPrefix(p.Repo, "oci://") && p.password != "" {
```

Comment:
IIRC, there are possibilities that the OCI repo may not start with `oci://` prefix. I'd suggest using the same OCI library as kpt here since that's more reliable.

Reply:
What kpt library are you referring too? Seems like kpt checks for "oci://" prefix:

https://github.com/GoogleContainerTools/kpt/blob/919b09de3b8c3a6663557a997a604f2379ae9232/internal/cmdreporeg/command.go#L108-L109

Reply:
Also I looked through the email thread again and CS said "Helm uses oci:// for pushing/pulling OCI charts". Maybe I misunderstood, but I thought that meant that we can safely use "oci://" prefix?

Reply:
kk. If this is aligned with kpt. I'm fine. 

I'm thinking of the oci module porch uses `github.com/google/go-containerregistry`.
For other format, I think container registry uses the format `https://{{REGION|ZONE}}-docker.pkg.dev/{{GCP_PROJECT}}/{{SUB_PATHS}}` 

I can't find the URL standards in the OCI repo though.

Reply:
Ah I'll look into `github.com/google/go-containerregistry` some more tomorrow, thanks for the tip

Reply:
It looks like helm doesn't do any better. Seems to be good enough for us 
https://github.com/helm/helm/blob/c137bfb68b4f8b7a10687f043b0dbccb09574969/pkg/registry/util.go#L37

Reply:
I can directly import helm's `IsOCI` function. At least then we will always be consistent with helm even if it changes.

Reply:
That will increase the size of the function (for builtin helm in kustomize, that increases the kustomize binary), not sure whether that worths it. Let's keep your original approach.